{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data input, cleaning and pre-processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading expression data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(WGCNA)\n",
    "options(stringsAsFactors = FALSE)\n",
    "femData = read.csv(\"LiverFemale3600.csv\")\n",
    "maleData = read.csv(\"LiverMale3600.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>3600</li>\n",
       "\t<li>143</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 3600\n",
       "\\item 143\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 3600\n",
       "2. 143\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 3600  143"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'substanceBXH'</li>\n",
       "\t<li>'gene_symbol'</li>\n",
       "\t<li>'LocusLinkID'</li>\n",
       "\t<li>'ProteomeID'</li>\n",
       "\t<li>'cytogeneticLoc'</li>\n",
       "\t<li>'CHROMOSOME'</li>\n",
       "\t<li>'StartPosition'</li>\n",
       "\t<li>'EndPosition'</li>\n",
       "\t<li>'F2_2'</li>\n",
       "\t<li>'F2_3'</li>\n",
       "\t<li>'F2_14'</li>\n",
       "\t<li>'F2_15'</li>\n",
       "\t<li>'F2_19'</li>\n",
       "\t<li>'F2_20'</li>\n",
       "\t<li>'F2_23'</li>\n",
       "\t<li>'F2_24'</li>\n",
       "\t<li>'F2_26'</li>\n",
       "\t<li>'F2_37'</li>\n",
       "\t<li>'F2_42'</li>\n",
       "\t<li>'F2_43'</li>\n",
       "\t<li>'F2_45'</li>\n",
       "\t<li>'F2_46'</li>\n",
       "\t<li>'F2_47'</li>\n",
       "\t<li>'F2_48'</li>\n",
       "\t<li>'F2_51'</li>\n",
       "\t<li>'F2_52'</li>\n",
       "\t<li>'F2_54'</li>\n",
       "\t<li>'F2_63'</li>\n",
       "\t<li>'F2_65'</li>\n",
       "\t<li>'F2_66'</li>\n",
       "\t<li>'F2_68'</li>\n",
       "\t<li>'F2_69'</li>\n",
       "\t<li>'F2_70'</li>\n",
       "\t<li>'F2_71'</li>\n",
       "\t<li>'F2_72'</li>\n",
       "\t<li>'F2_78'</li>\n",
       "\t<li>'F2_79'</li>\n",
       "\t<li>'F2_80'</li>\n",
       "\t<li>'F2_81'</li>\n",
       "\t<li>'F2_83'</li>\n",
       "\t<li>'F2_86'</li>\n",
       "\t<li>'F2_87'</li>\n",
       "\t<li>'F2_88'</li>\n",
       "\t<li>'F2_89'</li>\n",
       "\t<li>'F2_107'</li>\n",
       "\t<li>'F2_108'</li>\n",
       "\t<li>'F2_109'</li>\n",
       "\t<li>'F2_110'</li>\n",
       "\t<li>'F2_111'</li>\n",
       "\t<li>'F2_112'</li>\n",
       "\t<li>'F2_117'</li>\n",
       "\t<li>'F2_119'</li>\n",
       "\t<li>'F2_125'</li>\n",
       "\t<li>'F2_126'</li>\n",
       "\t<li>'F2_127'</li>\n",
       "\t<li>'F2_141'</li>\n",
       "\t<li>'F2_142'</li>\n",
       "\t<li>'F2_143'</li>\n",
       "\t<li>'F2_144'</li>\n",
       "\t<li>'F2_145'</li>\n",
       "\t<li>'F2_154'</li>\n",
       "\t<li>'F2_155'</li>\n",
       "\t<li>'F2_156'</li>\n",
       "\t<li>'F2_157'</li>\n",
       "\t<li>'F2_162'</li>\n",
       "\t<li>'F2_163'</li>\n",
       "\t<li>'F2_164'</li>\n",
       "\t<li>'F2_165'</li>\n",
       "\t<li>'F2_166'</li>\n",
       "\t<li>'F2_167'</li>\n",
       "\t<li>'F2_169'</li>\n",
       "\t<li>'F2_180'</li>\n",
       "\t<li>'F2_181'</li>\n",
       "\t<li>'F2_182'</li>\n",
       "\t<li>'F2_187'</li>\n",
       "\t<li>'F2_188'</li>\n",
       "\t<li>'F2_189'</li>\n",
       "\t<li>'F2_190'</li>\n",
       "\t<li>'F2_191'</li>\n",
       "\t<li>'F2_192'</li>\n",
       "\t<li>'F2_194'</li>\n",
       "\t<li>'F2_195'</li>\n",
       "\t<li>'F2_200'</li>\n",
       "\t<li>'F2_201'</li>\n",
       "\t<li>'F2_212'</li>\n",
       "\t<li>'F2_213'</li>\n",
       "\t<li>'F2_214'</li>\n",
       "\t<li>'F2_215'</li>\n",
       "\t<li>'F2_221'</li>\n",
       "\t<li>'F2_222'</li>\n",
       "\t<li>'F2_223'</li>\n",
       "\t<li>'F2_224'</li>\n",
       "\t<li>'F2_225'</li>\n",
       "\t<li>'F2_226'</li>\n",
       "\t<li>'F2_227'</li>\n",
       "\t<li>'F2_228'</li>\n",
       "\t<li>'F2_241'</li>\n",
       "\t<li>'F2_242'</li>\n",
       "\t<li>'F2_243'</li>\n",
       "\t<li>'F2_244'</li>\n",
       "\t<li>'F2_245'</li>\n",
       "\t<li>'F2_247'</li>\n",
       "\t<li>'F2_248'</li>\n",
       "\t<li>'F2_261'</li>\n",
       "\t<li>'F2_263'</li>\n",
       "\t<li>'F2_264'</li>\n",
       "\t<li>'F2_270'</li>\n",
       "\t<li>'F2_271'</li>\n",
       "\t<li>'F2_272'</li>\n",
       "\t<li>'F2_278'</li>\n",
       "\t<li>'F2_287'</li>\n",
       "\t<li>'F2_288'</li>\n",
       "\t<li>'F2_289'</li>\n",
       "\t<li>'F2_290'</li>\n",
       "\t<li>'F2_291'</li>\n",
       "\t<li>'F2_296'</li>\n",
       "\t<li>'F2_298'</li>\n",
       "\t<li>'F2_299'</li>\n",
       "\t<li>'F2_300'</li>\n",
       "\t<li>'F2_302'</li>\n",
       "\t<li>'F2_303'</li>\n",
       "\t<li>'F2_304'</li>\n",
       "\t<li>'F2_305'</li>\n",
       "\t<li>'F2_306'</li>\n",
       "\t<li>'F2_307'</li>\n",
       "\t<li>'F2_308'</li>\n",
       "\t<li>'F2_309'</li>\n",
       "\t<li>'F2_310'</li>\n",
       "\t<li>'F2_311'</li>\n",
       "\t<li>'F2_312'</li>\n",
       "\t<li>'F2_320'</li>\n",
       "\t<li>'F2_321'</li>\n",
       "\t<li>'F2_323'</li>\n",
       "\t<li>'F2_324'</li>\n",
       "\t<li>'F2_325'</li>\n",
       "\t<li>'F2_326'</li>\n",
       "\t<li>'F2_327'</li>\n",
       "\t<li>'F2_328'</li>\n",
       "\t<li>'F2_329'</li>\n",
       "\t<li>'F2_330'</li>\n",
       "\t<li>'F2_332'</li>\n",
       "\t<li>'F2_355'</li>\n",
       "\t<li>'F2_357'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'substanceBXH'\n",
       "\\item 'gene\\_symbol'\n",
       "\\item 'LocusLinkID'\n",
       "\\item 'ProteomeID'\n",
       "\\item 'cytogeneticLoc'\n",
       "\\item 'CHROMOSOME'\n",
       "\\item 'StartPosition'\n",
       "\\item 'EndPosition'\n",
       "\\item 'F2\\_2'\n",
       "\\item 'F2\\_3'\n",
       "\\item 'F2\\_14'\n",
       "\\item 'F2\\_15'\n",
       "\\item 'F2\\_19'\n",
       "\\item 'F2\\_20'\n",
       "\\item 'F2\\_23'\n",
       "\\item 'F2\\_24'\n",
       "\\item 'F2\\_26'\n",
       "\\item 'F2\\_37'\n",
       "\\item 'F2\\_42'\n",
       "\\item 'F2\\_43'\n",
       "\\item 'F2\\_45'\n",
       "\\item 'F2\\_46'\n",
       "\\item 'F2\\_47'\n",
       "\\item 'F2\\_48'\n",
       "\\item 'F2\\_51'\n",
       "\\item 'F2\\_52'\n",
       "\\item 'F2\\_54'\n",
       "\\item 'F2\\_63'\n",
       "\\item 'F2\\_65'\n",
       "\\item 'F2\\_66'\n",
       "\\item 'F2\\_68'\n",
       "\\item 'F2\\_69'\n",
       "\\item 'F2\\_70'\n",
       "\\item 'F2\\_71'\n",
       "\\item 'F2\\_72'\n",
       "\\item 'F2\\_78'\n",
       "\\item 'F2\\_79'\n",
       "\\item 'F2\\_80'\n",
       "\\item 'F2\\_81'\n",
       "\\item 'F2\\_83'\n",
       "\\item 'F2\\_86'\n",
       "\\item 'F2\\_87'\n",
       "\\item 'F2\\_88'\n",
       "\\item 'F2\\_89'\n",
       "\\item 'F2\\_107'\n",
       "\\item 'F2\\_108'\n",
       "\\item 'F2\\_109'\n",
       "\\item 'F2\\_110'\n",
       "\\item 'F2\\_111'\n",
       "\\item 'F2\\_112'\n",
       "\\item 'F2\\_117'\n",
       "\\item 'F2\\_119'\n",
       "\\item 'F2\\_125'\n",
       "\\item 'F2\\_126'\n",
       "\\item 'F2\\_127'\n",
       "\\item 'F2\\_141'\n",
       "\\item 'F2\\_142'\n",
       "\\item 'F2\\_143'\n",
       "\\item 'F2\\_144'\n",
       "\\item 'F2\\_145'\n",
       "\\item 'F2\\_154'\n",
       "\\item 'F2\\_155'\n",
       "\\item 'F2\\_156'\n",
       "\\item 'F2\\_157'\n",
       "\\item 'F2\\_162'\n",
       "\\item 'F2\\_163'\n",
       "\\item 'F2\\_164'\n",
       "\\item 'F2\\_165'\n",
       "\\item 'F2\\_166'\n",
       "\\item 'F2\\_167'\n",
       "\\item 'F2\\_169'\n",
       "\\item 'F2\\_180'\n",
       "\\item 'F2\\_181'\n",
       "\\item 'F2\\_182'\n",
       "\\item 'F2\\_187'\n",
       "\\item 'F2\\_188'\n",
       "\\item 'F2\\_189'\n",
       "\\item 'F2\\_190'\n",
       "\\item 'F2\\_191'\n",
       "\\item 'F2\\_192'\n",
       "\\item 'F2\\_194'\n",
       "\\item 'F2\\_195'\n",
       "\\item 'F2\\_200'\n",
       "\\item 'F2\\_201'\n",
       "\\item 'F2\\_212'\n",
       "\\item 'F2\\_213'\n",
       "\\item 'F2\\_214'\n",
       "\\item 'F2\\_215'\n",
       "\\item 'F2\\_221'\n",
       "\\item 'F2\\_222'\n",
       "\\item 'F2\\_223'\n",
       "\\item 'F2\\_224'\n",
       "\\item 'F2\\_225'\n",
       "\\item 'F2\\_226'\n",
       "\\item 'F2\\_227'\n",
       "\\item 'F2\\_228'\n",
       "\\item 'F2\\_241'\n",
       "\\item 'F2\\_242'\n",
       "\\item 'F2\\_243'\n",
       "\\item 'F2\\_244'\n",
       "\\item 'F2\\_245'\n",
       "\\item 'F2\\_247'\n",
       "\\item 'F2\\_248'\n",
       "\\item 'F2\\_261'\n",
       "\\item 'F2\\_263'\n",
       "\\item 'F2\\_264'\n",
       "\\item 'F2\\_270'\n",
       "\\item 'F2\\_271'\n",
       "\\item 'F2\\_272'\n",
       "\\item 'F2\\_278'\n",
       "\\item 'F2\\_287'\n",
       "\\item 'F2\\_288'\n",
       "\\item 'F2\\_289'\n",
       "\\item 'F2\\_290'\n",
       "\\item 'F2\\_291'\n",
       "\\item 'F2\\_296'\n",
       "\\item 'F2\\_298'\n",
       "\\item 'F2\\_299'\n",
       "\\item 'F2\\_300'\n",
       "\\item 'F2\\_302'\n",
       "\\item 'F2\\_303'\n",
       "\\item 'F2\\_304'\n",
       "\\item 'F2\\_305'\n",
       "\\item 'F2\\_306'\n",
       "\\item 'F2\\_307'\n",
       "\\item 'F2\\_308'\n",
       "\\item 'F2\\_309'\n",
       "\\item 'F2\\_310'\n",
       "\\item 'F2\\_311'\n",
       "\\item 'F2\\_312'\n",
       "\\item 'F2\\_320'\n",
       "\\item 'F2\\_321'\n",
       "\\item 'F2\\_323'\n",
       "\\item 'F2\\_324'\n",
       "\\item 'F2\\_325'\n",
       "\\item 'F2\\_326'\n",
       "\\item 'F2\\_327'\n",
       "\\item 'F2\\_328'\n",
       "\\item 'F2\\_329'\n",
       "\\item 'F2\\_330'\n",
       "\\item 'F2\\_332'\n",
       "\\item 'F2\\_355'\n",
       "\\item 'F2\\_357'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'substanceBXH'\n",
       "2. 'gene_symbol'\n",
       "3. 'LocusLinkID'\n",
       "4. 'ProteomeID'\n",
       "5. 'cytogeneticLoc'\n",
       "6. 'CHROMOSOME'\n",
       "7. 'StartPosition'\n",
       "8. 'EndPosition'\n",
       "9. 'F2_2'\n",
       "10. 'F2_3'\n",
       "11. 'F2_14'\n",
       "12. 'F2_15'\n",
       "13. 'F2_19'\n",
       "14. 'F2_20'\n",
       "15. 'F2_23'\n",
       "16. 'F2_24'\n",
       "17. 'F2_26'\n",
       "18. 'F2_37'\n",
       "19. 'F2_42'\n",
       "20. 'F2_43'\n",
       "21. 'F2_45'\n",
       "22. 'F2_46'\n",
       "23. 'F2_47'\n",
       "24. 'F2_48'\n",
       "25. 'F2_51'\n",
       "26. 'F2_52'\n",
       "27. 'F2_54'\n",
       "28. 'F2_63'\n",
       "29. 'F2_65'\n",
       "30. 'F2_66'\n",
       "31. 'F2_68'\n",
       "32. 'F2_69'\n",
       "33. 'F2_70'\n",
       "34. 'F2_71'\n",
       "35. 'F2_72'\n",
       "36. 'F2_78'\n",
       "37. 'F2_79'\n",
       "38. 'F2_80'\n",
       "39. 'F2_81'\n",
       "40. 'F2_83'\n",
       "41. 'F2_86'\n",
       "42. 'F2_87'\n",
       "43. 'F2_88'\n",
       "44. 'F2_89'\n",
       "45. 'F2_107'\n",
       "46. 'F2_108'\n",
       "47. 'F2_109'\n",
       "48. 'F2_110'\n",
       "49. 'F2_111'\n",
       "50. 'F2_112'\n",
       "51. 'F2_117'\n",
       "52. 'F2_119'\n",
       "53. 'F2_125'\n",
       "54. 'F2_126'\n",
       "55. 'F2_127'\n",
       "56. 'F2_141'\n",
       "57. 'F2_142'\n",
       "58. 'F2_143'\n",
       "59. 'F2_144'\n",
       "60. 'F2_145'\n",
       "61. 'F2_154'\n",
       "62. 'F2_155'\n",
       "63. 'F2_156'\n",
       "64. 'F2_157'\n",
       "65. 'F2_162'\n",
       "66. 'F2_163'\n",
       "67. 'F2_164'\n",
       "68. 'F2_165'\n",
       "69. 'F2_166'\n",
       "70. 'F2_167'\n",
       "71. 'F2_169'\n",
       "72. 'F2_180'\n",
       "73. 'F2_181'\n",
       "74. 'F2_182'\n",
       "75. 'F2_187'\n",
       "76. 'F2_188'\n",
       "77. 'F2_189'\n",
       "78. 'F2_190'\n",
       "79. 'F2_191'\n",
       "80. 'F2_192'\n",
       "81. 'F2_194'\n",
       "82. 'F2_195'\n",
       "83. 'F2_200'\n",
       "84. 'F2_201'\n",
       "85. 'F2_212'\n",
       "86. 'F2_213'\n",
       "87. 'F2_214'\n",
       "88. 'F2_215'\n",
       "89. 'F2_221'\n",
       "90. 'F2_222'\n",
       "91. 'F2_223'\n",
       "92. 'F2_224'\n",
       "93. 'F2_225'\n",
       "94. 'F2_226'\n",
       "95. 'F2_227'\n",
       "96. 'F2_228'\n",
       "97. 'F2_241'\n",
       "98. 'F2_242'\n",
       "99. 'F2_243'\n",
       "100. 'F2_244'\n",
       "101. 'F2_245'\n",
       "102. 'F2_247'\n",
       "103. 'F2_248'\n",
       "104. 'F2_261'\n",
       "105. 'F2_263'\n",
       "106. 'F2_264'\n",
       "107. 'F2_270'\n",
       "108. 'F2_271'\n",
       "109. 'F2_272'\n",
       "110. 'F2_278'\n",
       "111. 'F2_287'\n",
       "112. 'F2_288'\n",
       "113. 'F2_289'\n",
       "114. 'F2_290'\n",
       "115. 'F2_291'\n",
       "116. 'F2_296'\n",
       "117. 'F2_298'\n",
       "118. 'F2_299'\n",
       "119. 'F2_300'\n",
       "120. 'F2_302'\n",
       "121. 'F2_303'\n",
       "122. 'F2_304'\n",
       "123. 'F2_305'\n",
       "124. 'F2_306'\n",
       "125. 'F2_307'\n",
       "126. 'F2_308'\n",
       "127. 'F2_309'\n",
       "128. 'F2_310'\n",
       "129. 'F2_311'\n",
       "130. 'F2_312'\n",
       "131. 'F2_320'\n",
       "132. 'F2_321'\n",
       "133. 'F2_323'\n",
       "134. 'F2_324'\n",
       "135. 'F2_325'\n",
       "136. 'F2_326'\n",
       "137. 'F2_327'\n",
       "138. 'F2_328'\n",
       "139. 'F2_329'\n",
       "140. 'F2_330'\n",
       "141. 'F2_332'\n",
       "142. 'F2_355'\n",
       "143. 'F2_357'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"substanceBXH\"   \"gene_symbol\"    \"LocusLinkID\"    \"ProteomeID\"    \n",
       "  [5] \"cytogeneticLoc\" \"CHROMOSOME\"     \"StartPosition\"  \"EndPosition\"   \n",
       "  [9] \"F2_2\"           \"F2_3\"           \"F2_14\"          \"F2_15\"         \n",
       " [13] \"F2_19\"          \"F2_20\"          \"F2_23\"          \"F2_24\"         \n",
       " [17] \"F2_26\"          \"F2_37\"          \"F2_42\"          \"F2_43\"         \n",
       " [21] \"F2_45\"          \"F2_46\"          \"F2_47\"          \"F2_48\"         \n",
       " [25] \"F2_51\"          \"F2_52\"          \"F2_54\"          \"F2_63\"         \n",
       " [29] \"F2_65\"          \"F2_66\"          \"F2_68\"          \"F2_69\"         \n",
       " [33] \"F2_70\"          \"F2_71\"          \"F2_72\"          \"F2_78\"         \n",
       " [37] \"F2_79\"          \"F2_80\"          \"F2_81\"          \"F2_83\"         \n",
       " [41] \"F2_86\"          \"F2_87\"          \"F2_88\"          \"F2_89\"         \n",
       " [45] \"F2_107\"         \"F2_108\"         \"F2_109\"         \"F2_110\"        \n",
       " [49] \"F2_111\"         \"F2_112\"         \"F2_117\"         \"F2_119\"        \n",
       " [53] \"F2_125\"         \"F2_126\"         \"F2_127\"         \"F2_141\"        \n",
       " [57] \"F2_142\"         \"F2_143\"         \"F2_144\"         \"F2_145\"        \n",
       " [61] \"F2_154\"         \"F2_155\"         \"F2_156\"         \"F2_157\"        \n",
       " [65] \"F2_162\"         \"F2_163\"         \"F2_164\"         \"F2_165\"        \n",
       " [69] \"F2_166\"         \"F2_167\"         \"F2_169\"         \"F2_180\"        \n",
       " [73] \"F2_181\"         \"F2_182\"         \"F2_187\"         \"F2_188\"        \n",
       " [77] \"F2_189\"         \"F2_190\"         \"F2_191\"         \"F2_192\"        \n",
       " [81] \"F2_194\"         \"F2_195\"         \"F2_200\"         \"F2_201\"        \n",
       " [85] \"F2_212\"         \"F2_213\"         \"F2_214\"         \"F2_215\"        \n",
       " [89] \"F2_221\"         \"F2_222\"         \"F2_223\"         \"F2_224\"        \n",
       " [93] \"F2_225\"         \"F2_226\"         \"F2_227\"         \"F2_228\"        \n",
       " [97] \"F2_241\"         \"F2_242\"         \"F2_243\"         \"F2_244\"        \n",
       "[101] \"F2_245\"         \"F2_247\"         \"F2_248\"         \"F2_261\"        \n",
       "[105] \"F2_263\"         \"F2_264\"         \"F2_270\"         \"F2_271\"        \n",
       "[109] \"F2_272\"         \"F2_278\"         \"F2_287\"         \"F2_288\"        \n",
       "[113] \"F2_289\"         \"F2_290\"         \"F2_291\"         \"F2_296\"        \n",
       "[117] \"F2_298\"         \"F2_299\"         \"F2_300\"         \"F2_302\"        \n",
       "[121] \"F2_303\"         \"F2_304\"         \"F2_305\"         \"F2_306\"        \n",
       "[125] \"F2_307\"         \"F2_308\"         \"F2_309\"         \"F2_310\"        \n",
       "[129] \"F2_311\"         \"F2_312\"         \"F2_320\"         \"F2_321\"        \n",
       "[133] \"F2_323\"         \"F2_324\"         \"F2_325\"         \"F2_326\"        \n",
       "[137] \"F2_327\"         \"F2_328\"         \"F2_329\"         \"F2_330\"        \n",
       "[141] \"F2_332\"         \"F2_355\"         \"F2_357\"        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(femData)\n",
    "names(femData)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "nSets = 2\n",
    "setLabels = c(\"Female liver\", \"Male liver\")\n",
    "shortLabels = c(\"Female\", \"Male\")\n",
    "\n",
    "multiExpr = vector(mode = \"list\", length = nSets)\n",
    "multiExpr[[1]] = list(data = as.data.frame(t(femData[-c(1:8)]))) # name is \"data\"\n",
    "names(multiExpr[[1]]$data) = femData$substanceBXH # feature name\n",
    "rownames(multiExpr[[1]]$data) = names(femData)[-c(1:8)] # name of samples\n",
    "multiExpr[[2]] = list(data = as.data.frame(t(maleData[-c(1:8)])))\n",
    "names(multiExpr[[2]]$data) = maleData$substanceBXH;\n",
    "rownames(multiExpr[[2]]$data) = names(maleData)[-c(1:8)];\n",
    "exprSize = checkSets(multiExpr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$nSets</dt>\n",
       "\t\t<dd>2</dd>\n",
       "\t<dt>$nGenes</dt>\n",
       "\t\t<dd>3600</dd>\n",
       "\t<dt>$nSamples</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>135</li>\n",
       "\t<li>124</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$structureOK</dt>\n",
       "\t\t<dd>TRUE</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$nSets] 2\n",
       "\\item[\\$nGenes] 3600\n",
       "\\item[\\$nSamples] \\begin{enumerate*}\n",
       "\\item 135\n",
       "\\item 124\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$structureOK] TRUE\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$nSets\n",
       ":   2\n",
       "$nGenes\n",
       ":   3600\n",
       "$nSamples\n",
       ":   1. 135\n",
       "2. 124\n",
       "\n",
       "\n",
       "\n",
       "$structureOK\n",
       ":   TRUE\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$nSets\n",
       "[1] 2\n",
       "\n",
       "$nGenes\n",
       "[1] 3600\n",
       "\n",
       "$nSamples\n",
       "[1] 135 124\n",
       "\n",
       "$structureOK\n",
       "[1] TRUE\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "exprSize"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Rudimentary data cleaning and outlier removal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Flagging genes and samples with too many missing values...\n",
      "  ..step 1\n",
      "   ..bad gene count: 0, bad sample counts: 0, 0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Check that all genes and samples have sufficiently low numbers of missing values.\n",
    "gsg = goodSamplesGenesMS(multiExpr, verbose = 3)\n",
    "gsg$allOK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "sampleTrees = list()\n",
    "for (set in 1:nSets){\n",
    "    sampleTrees[[set]] = hclust(dist(multiExpr[[set]]$data), method = \"average\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pdf(file = \"Plots_SampleClustering.pdf\", width = 12, height = 12)\n",
    "par(mfrow=c(2,1))\n",
    "par(mar = c(0, 4, 2, 0))\n",
    "for (set in 1:nSets){\n",
    "    plot(sampleTrees[[set]], \n",
    "         main = paste(\"Sample clustering on all genes in\", setLabels[set]),\n",
    "         xlab=\"\", sub=\"\", cex = 0.7)\n",
    "}\n",
    "dev.off() #device.off"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#There is an outliers\n",
    "# Choose the \"base\" cut height for the female data set\n",
    "baseHeight = 16\n",
    "cutHeights = c(16, 16*exprSize$nSamples[2]/exprSize$nSamples[1])\n",
    "# Re-plot the dendrograms including the cut lines\n",
    "pdf(file = \"Plots_SampleClustering.pdf\", width = 12, height = 12)\n",
    "par(mfrow=c(2,1))\n",
    "par(mar = c(0, 4, 2, 0))\n",
    "for (set in 1:nSets){\n",
    "    plot(sampleTrees[[set]], \n",
    "         main = paste(\"Sample clustering on all genes in\", setLabels[set]),\n",
    "         xlab=\"\", sub=\"\", cex = 0.7)\n",
    "    abline(h=cutHeights[set], col = \"red\")\n",
    "}\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$nSets</dt>\n",
       "\t\t<dd>2</dd>\n",
       "\t<dt>$nGenes</dt>\n",
       "\t\t<dd>3600</dd>\n",
       "\t<dt>$nSamples</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>134</li>\n",
       "\t<li>124</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$structureOK</dt>\n",
       "\t\t<dd>TRUE</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$nSets] 2\n",
       "\\item[\\$nGenes] 3600\n",
       "\\item[\\$nSamples] \\begin{enumerate*}\n",
       "\\item 134\n",
       "\\item 124\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$structureOK] TRUE\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$nSets\n",
       ":   2\n",
       "$nGenes\n",
       ":   3600\n",
       "$nSamples\n",
       ":   1. 134\n",
       "2. 124\n",
       "\n",
       "\n",
       "\n",
       "$structureOK\n",
       ":   TRUE\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$nSets\n",
       "[1] 2\n",
       "\n",
       "$nGenes\n",
       "[1] 3600\n",
       "\n",
       "$nSamples\n",
       "[1] 134 124\n",
       "\n",
       "$structureOK\n",
       "[1] TRUE\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#  outlier removal\n",
    "for (set in 1:nSets){\n",
    "    # Find clusters cut by the line\n",
    "    labels = cutreeStatic(sampleTrees[[set]], cutHeight = cutHeights[set])\n",
    "    labels\n",
    "    # Keep the largest one (labeled by the number 1)\n",
    "    keep = (labels==1)\n",
    "    multiExpr[[set]]$data = multiExpr[[set]]$data[keep, ]\n",
    "}\n",
    "collectGarbage();\n",
    "\n",
    "# Check the size of the leftover data\n",
    "exprSize = checkSets(multiExpr)\n",
    "exprSize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. TRUE\n",
       "2. TRUE\n",
       "3. TRUE\n",
       "4. TRUE\n",
       "5. TRUE\n",
       "6. TRUE\n",
       "7. TRUE\n",
       "8. TRUE\n",
       "9. TRUE\n",
       "10. TRUE\n",
       "11. TRUE\n",
       "12. TRUE\n",
       "13. TRUE\n",
       "14. TRUE\n",
       "15. TRUE\n",
       "16. TRUE\n",
       "17. TRUE\n",
       "18. TRUE\n",
       "19. TRUE\n",
       "20. TRUE\n",
       "21. TRUE\n",
       "22. TRUE\n",
       "23. TRUE\n",
       "24. TRUE\n",
       "25. TRUE\n",
       "26. TRUE\n",
       "27. TRUE\n",
       "28. TRUE\n",
       "29. TRUE\n",
       "30. TRUE\n",
       "31. TRUE\n",
       "32. TRUE\n",
       "33. TRUE\n",
       "34. TRUE\n",
       "35. TRUE\n",
       "36. TRUE\n",
       "37. TRUE\n",
       "38. TRUE\n",
       "39. TRUE\n",
       "40. TRUE\n",
       "41. TRUE\n",
       "42. TRUE\n",
       "43. TRUE\n",
       "44. TRUE\n",
       "45. TRUE\n",
       "46. TRUE\n",
       "47. TRUE\n",
       "48. TRUE\n",
       "49. TRUE\n",
       "50. TRUE\n",
       "51. TRUE\n",
       "52. TRUE\n",
       "53. TRUE\n",
       "54. TRUE\n",
       "55. TRUE\n",
       "56. TRUE\n",
       "57. TRUE\n",
       "58. TRUE\n",
       "59. TRUE\n",
       "60. TRUE\n",
       "61. TRUE\n",
       "62. TRUE\n",
       "63. TRUE\n",
       "64. TRUE\n",
       "65. TRUE\n",
       "66. TRUE\n",
       "67. TRUE\n",
       "68. TRUE\n",
       "69. TRUE\n",
       "70. TRUE\n",
       "71. TRUE\n",
       "72. TRUE\n",
       "73. TRUE\n",
       "74. TRUE\n",
       "75. TRUE\n",
       "76. TRUE\n",
       "77. TRUE\n",
       "78. TRUE\n",
       "79. TRUE\n",
       "80. TRUE\n",
       "81. FALSE\n",
       "82. TRUE\n",
       "83. TRUE\n",
       "84. TRUE\n",
       "85. TRUE\n",
       "86. TRUE\n",
       "87. TRUE\n",
       "88. TRUE\n",
       "89. TRUE\n",
       "90. TRUE\n",
       "91. TRUE\n",
       "92. TRUE\n",
       "93. TRUE\n",
       "94. TRUE\n",
       "95. TRUE\n",
       "96. TRUE\n",
       "97. TRUE\n",
       "98. TRUE\n",
       "99. TRUE\n",
       "100. TRUE\n",
       "101. TRUE\n",
       "102. TRUE\n",
       "103. TRUE\n",
       "104. TRUE\n",
       "105. TRUE\n",
       "106. TRUE\n",
       "107. TRUE\n",
       "108. TRUE\n",
       "109. TRUE\n",
       "110. TRUE\n",
       "111. TRUE\n",
       "112. TRUE\n",
       "113. TRUE\n",
       "114. TRUE\n",
       "115. TRUE\n",
       "116. TRUE\n",
       "117. TRUE\n",
       "118. TRUE\n",
       "119. TRUE\n",
       "120. TRUE\n",
       "121. TRUE\n",
       "122. TRUE\n",
       "123. TRUE\n",
       "124. TRUE\n",
       "125. TRUE\n",
       "126. TRUE\n",
       "127. TRUE\n",
       "128. TRUE\n",
       "129. TRUE\n",
       "130. TRUE\n",
       "131. TRUE\n",
       "132. TRUE\n",
       "133. TRUE\n",
       "134. TRUE\n",
       "135. TRUE\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       " [13]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       " [25]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       " [37]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       " [49]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       " [61]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       " [73]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE\n",
       " [85]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       " [97]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       "[109]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       "[121]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       "[133]  TRUE  TRUE  TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cutreeStatic(sampleTrees[[1]], cutHeight = cutHeights[1])==1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading clinical trait data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'X'</li>\n",
       "\t<li>'Mice'</li>\n",
       "\t<li>'Number'</li>\n",
       "\t<li>'Mouse_ID'</li>\n",
       "\t<li>'Strain'</li>\n",
       "\t<li>'sex'</li>\n",
       "\t<li>'DOB'</li>\n",
       "\t<li>'parents'</li>\n",
       "\t<li>'Western_Diet'</li>\n",
       "\t<li>'Sac_Date'</li>\n",
       "\t<li>'weight_g'</li>\n",
       "\t<li>'length_cm'</li>\n",
       "\t<li>'ab_fat'</li>\n",
       "\t<li>'other_fat'</li>\n",
       "\t<li>'total_fat'</li>\n",
       "\t<li>'comments'</li>\n",
       "\t<li>'X100xfat_weight'</li>\n",
       "\t<li>'Trigly'</li>\n",
       "\t<li>'Total_Chol'</li>\n",
       "\t<li>'HDL_Chol'</li>\n",
       "\t<li>'UC'</li>\n",
       "\t<li>'FFA'</li>\n",
       "\t<li>'Glucose'</li>\n",
       "\t<li>'LDL_plus_VLDL'</li>\n",
       "\t<li>'MCP_1_phys'</li>\n",
       "\t<li>'Insulin_ug_l'</li>\n",
       "\t<li>'Glucose_Insulin'</li>\n",
       "\t<li>'Leptin_pg_ml'</li>\n",
       "\t<li>'Adiponectin'</li>\n",
       "\t<li>'Aortic.lesions'</li>\n",
       "\t<li>'Note'</li>\n",
       "\t<li>'Aneurysm'</li>\n",
       "\t<li>'Aortic_cal_M'</li>\n",
       "\t<li>'Aortic_cal_L'</li>\n",
       "\t<li>'CoronaryArtery_Cal'</li>\n",
       "\t<li>'Myocardial_cal'</li>\n",
       "\t<li>'BMD_all_limbs'</li>\n",
       "\t<li>'BMD_femurs_only'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'X'\n",
       "\\item 'Mice'\n",
       "\\item 'Number'\n",
       "\\item 'Mouse\\_ID'\n",
       "\\item 'Strain'\n",
       "\\item 'sex'\n",
       "\\item 'DOB'\n",
       "\\item 'parents'\n",
       "\\item 'Western\\_Diet'\n",
       "\\item 'Sac\\_Date'\n",
       "\\item 'weight\\_g'\n",
       "\\item 'length\\_cm'\n",
       "\\item 'ab\\_fat'\n",
       "\\item 'other\\_fat'\n",
       "\\item 'total\\_fat'\n",
       "\\item 'comments'\n",
       "\\item 'X100xfat\\_weight'\n",
       "\\item 'Trigly'\n",
       "\\item 'Total\\_Chol'\n",
       "\\item 'HDL\\_Chol'\n",
       "\\item 'UC'\n",
       "\\item 'FFA'\n",
       "\\item 'Glucose'\n",
       "\\item 'LDL\\_plus\\_VLDL'\n",
       "\\item 'MCP\\_1\\_phys'\n",
       "\\item 'Insulin\\_ug\\_l'\n",
       "\\item 'Glucose\\_Insulin'\n",
       "\\item 'Leptin\\_pg\\_ml'\n",
       "\\item 'Adiponectin'\n",
       "\\item 'Aortic.lesions'\n",
       "\\item 'Note'\n",
       "\\item 'Aneurysm'\n",
       "\\item 'Aortic\\_cal\\_M'\n",
       "\\item 'Aortic\\_cal\\_L'\n",
       "\\item 'CoronaryArtery\\_Cal'\n",
       "\\item 'Myocardial\\_cal'\n",
       "\\item 'BMD\\_all\\_limbs'\n",
       "\\item 'BMD\\_femurs\\_only'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'X'\n",
       "2. 'Mice'\n",
       "3. 'Number'\n",
       "4. 'Mouse_ID'\n",
       "5. 'Strain'\n",
       "6. 'sex'\n",
       "7. 'DOB'\n",
       "8. 'parents'\n",
       "9. 'Western_Diet'\n",
       "10. 'Sac_Date'\n",
       "11. 'weight_g'\n",
       "12. 'length_cm'\n",
       "13. 'ab_fat'\n",
       "14. 'other_fat'\n",
       "15. 'total_fat'\n",
       "16. 'comments'\n",
       "17. 'X100xfat_weight'\n",
       "18. 'Trigly'\n",
       "19. 'Total_Chol'\n",
       "20. 'HDL_Chol'\n",
       "21. 'UC'\n",
       "22. 'FFA'\n",
       "23. 'Glucose'\n",
       "24. 'LDL_plus_VLDL'\n",
       "25. 'MCP_1_phys'\n",
       "26. 'Insulin_ug_l'\n",
       "27. 'Glucose_Insulin'\n",
       "28. 'Leptin_pg_ml'\n",
       "29. 'Adiponectin'\n",
       "30. 'Aortic.lesions'\n",
       "31. 'Note'\n",
       "32. 'Aneurysm'\n",
       "33. 'Aortic_cal_M'\n",
       "34. 'Aortic_cal_L'\n",
       "35. 'CoronaryArtery_Cal'\n",
       "36. 'Myocardial_cal'\n",
       "37. 'BMD_all_limbs'\n",
       "38. 'BMD_femurs_only'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"X\"                  \"Mice\"               \"Number\"            \n",
       " [4] \"Mouse_ID\"           \"Strain\"             \"sex\"               \n",
       " [7] \"DOB\"                \"parents\"            \"Western_Diet\"      \n",
       "[10] \"Sac_Date\"           \"weight_g\"           \"length_cm\"         \n",
       "[13] \"ab_fat\"             \"other_fat\"          \"total_fat\"         \n",
       "[16] \"comments\"           \"X100xfat_weight\"    \"Trigly\"            \n",
       "[19] \"Total_Chol\"         \"HDL_Chol\"           \"UC\"                \n",
       "[22] \"FFA\"                \"Glucose\"            \"LDL_plus_VLDL\"     \n",
       "[25] \"MCP_1_phys\"         \"Insulin_ug_l\"       \"Glucose_Insulin\"   \n",
       "[28] \"Leptin_pg_ml\"       \"Adiponectin\"        \"Aortic.lesions\"    \n",
       "[31] \"Note\"               \"Aneurysm\"           \"Aortic_cal_M\"      \n",
       "[34] \"Aortic_cal_L\"       \"CoronaryArtery_Cal\" \"Myocardial_cal\"    \n",
       "[37] \"BMD_all_limbs\"      \"BMD_femurs_only\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>361</li>\n",
       "\t<li>38</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 361\n",
       "\\item 38\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 361\n",
       "2. 38\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 361  38"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "traitData = read.csv(\"ClinicalTraits.csv\")\n",
    "names(traitData)\n",
    "dim(traitData)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>361</li>\n",
       "\t<li>27</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 361\n",
       "\\item 27\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 361\n",
       "2. 27\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 361  27"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'Mice'</li>\n",
       "\t<li>'weight_g'</li>\n",
       "\t<li>'length_cm'</li>\n",
       "\t<li>'ab_fat'</li>\n",
       "\t<li>'other_fat'</li>\n",
       "\t<li>'total_fat'</li>\n",
       "\t<li>'X100xfat_weight'</li>\n",
       "\t<li>'Trigly'</li>\n",
       "\t<li>'Total_Chol'</li>\n",
       "\t<li>'HDL_Chol'</li>\n",
       "\t<li>'UC'</li>\n",
       "\t<li>'FFA'</li>\n",
       "\t<li>'Glucose'</li>\n",
       "\t<li>'LDL_plus_VLDL'</li>\n",
       "\t<li>'MCP_1_phys'</li>\n",
       "\t<li>'Insulin_ug_l'</li>\n",
       "\t<li>'Glucose_Insulin'</li>\n",
       "\t<li>'Leptin_pg_ml'</li>\n",
       "\t<li>'Adiponectin'</li>\n",
       "\t<li>'Aortic.lesions'</li>\n",
       "\t<li>'Aneurysm'</li>\n",
       "\t<li>'Aortic_cal_M'</li>\n",
       "\t<li>'Aortic_cal_L'</li>\n",
       "\t<li>'CoronaryArtery_Cal'</li>\n",
       "\t<li>'Myocardial_cal'</li>\n",
       "\t<li>'BMD_all_limbs'</li>\n",
       "\t<li>'BMD_femurs_only'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Mice'\n",
       "\\item 'weight\\_g'\n",
       "\\item 'length\\_cm'\n",
       "\\item 'ab\\_fat'\n",
       "\\item 'other\\_fat'\n",
       "\\item 'total\\_fat'\n",
       "\\item 'X100xfat\\_weight'\n",
       "\\item 'Trigly'\n",
       "\\item 'Total\\_Chol'\n",
       "\\item 'HDL\\_Chol'\n",
       "\\item 'UC'\n",
       "\\item 'FFA'\n",
       "\\item 'Glucose'\n",
       "\\item 'LDL\\_plus\\_VLDL'\n",
       "\\item 'MCP\\_1\\_phys'\n",
       "\\item 'Insulin\\_ug\\_l'\n",
       "\\item 'Glucose\\_Insulin'\n",
       "\\item 'Leptin\\_pg\\_ml'\n",
       "\\item 'Adiponectin'\n",
       "\\item 'Aortic.lesions'\n",
       "\\item 'Aneurysm'\n",
       "\\item 'Aortic\\_cal\\_M'\n",
       "\\item 'Aortic\\_cal\\_L'\n",
       "\\item 'CoronaryArtery\\_Cal'\n",
       "\\item 'Myocardial\\_cal'\n",
       "\\item 'BMD\\_all\\_limbs'\n",
       "\\item 'BMD\\_femurs\\_only'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Mice'\n",
       "2. 'weight_g'\n",
       "3. 'length_cm'\n",
       "4. 'ab_fat'\n",
       "5. 'other_fat'\n",
       "6. 'total_fat'\n",
       "7. 'X100xfat_weight'\n",
       "8. 'Trigly'\n",
       "9. 'Total_Chol'\n",
       "10. 'HDL_Chol'\n",
       "11. 'UC'\n",
       "12. 'FFA'\n",
       "13. 'Glucose'\n",
       "14. 'LDL_plus_VLDL'\n",
       "15. 'MCP_1_phys'\n",
       "16. 'Insulin_ug_l'\n",
       "17. 'Glucose_Insulin'\n",
       "18. 'Leptin_pg_ml'\n",
       "19. 'Adiponectin'\n",
       "20. 'Aortic.lesions'\n",
       "21. 'Aneurysm'\n",
       "22. 'Aortic_cal_M'\n",
       "23. 'Aortic_cal_L'\n",
       "24. 'CoronaryArtery_Cal'\n",
       "25. 'Myocardial_cal'\n",
       "26. 'BMD_all_limbs'\n",
       "27. 'BMD_femurs_only'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Mice\"               \"weight_g\"           \"length_cm\"         \n",
       " [4] \"ab_fat\"             \"other_fat\"          \"total_fat\"         \n",
       " [7] \"X100xfat_weight\"    \"Trigly\"             \"Total_Chol\"        \n",
       "[10] \"HDL_Chol\"           \"UC\"                 \"FFA\"               \n",
       "[13] \"Glucose\"            \"LDL_plus_VLDL\"      \"MCP_1_phys\"        \n",
       "[16] \"Insulin_ug_l\"       \"Glucose_Insulin\"    \"Leptin_pg_ml\"      \n",
       "[19] \"Adiponectin\"        \"Aortic.lesions\"     \"Aneurysm\"          \n",
       "[22] \"Aortic_cal_M\"       \"Aortic_cal_L\"       \"CoronaryArtery_Cal\"\n",
       "[25] \"Myocardial_cal\"     \"BMD_all_limbs\"      \"BMD_femurs_only\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# remove columns that hold information we do not need.\n",
    "allTraits = traitData[, -c(31, 16)]\n",
    "allTraits = allTraits[, c(2, 11:36) ]\n",
    "dim(allTraits)\n",
    "names(allTraits)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Form a multi-set structure that will hold the clinical traits.\n",
    "Traits = vector(mode=\"list\", length = nSets)\n",
    "for (set in 1:nSets){\n",
    "    setSamples = rownames(multiExpr[[set]]$data)\n",
    "    traitRows = match(setSamples, allTraits$Mice) # find a correspondance between rows\n",
    "    Traits[[set]] = list(data = allTraits[traitRows, -1])\n",
    "    rownames(Traits[[set]]$data) = allTraits[traitRows, 1]\n",
    "}\n",
    "collectGarbage()\n",
    "# Define data set dimensions\n",
    "nGenes = exprSize$nGenes\n",
    "nSamples = exprSize$nSamples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>134</li>\n",
       "\t<li>124</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 134\n",
       "\\item 124\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 134\n",
       "2. 124\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 134 124"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nSamples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "save(multiExpr, Traits, nGenes, nSamples, setLabels, \n",
    "     shortLabels, exprSize,file = \"Consensus-dataInput.RData\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
